blockwork/build/interface.py

```diff
@@ -185,9 +172,9 @@ def resolve_meta(self, fn):
 
 
 class FileInterface(Interface[Path]):
-    def resolve_container(self, ctx: "Context", container: Container):
+    def resolve_container(self, ctx: "Context", container: Container, direction: Interface.Direction):
         host_path = self.resolve(ctx)
         container_path = ctx.map_to_container(host_path)
-        readonly = self.direction is InterfaceDirection.Input
+        readonly = direction is self.Direction.Input
```

Comment:
Yeah I find `self.Direction` a bit weird here - because `Direction` is not an instance member, it's a class variable. So either `Instance.Direction` or `InstanceDirection` (if the enum is taken out of the class namespace)